File: pyg-manager/pyg-manager-interface/src/main/java/com/pyg/maneger/service/specificationService.java
package com.pyg.maneger.service;

import com.pyg.pojo.TbSpecification;
import com.pyg.vo.Specification;
import com.pyg.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/8/10.
 */
public interface specificationService {


    //分页查询
    PageResult findPage(int pageNum, int pageSize);
    //保存品牌
   void save(Specification specification);
    //根据id查询
    Specification findById(Long id);
    //根据id修改
    void updata(Specification specification);
    //根据id删除
    void del(Long[] ids);
    //新增
    List<Map> findspecIdsList();

}

File: pyg-shop-web/src/main/webapp/js/service/specificationService.js
//定义模块函数
app.service('specificationService',function ($http) {
    //查询全部
    this.findAll=function () {
        return  $http.get('../specification/findAll')
    };
    //分页查询
    this.findPage=function (pageNum, pageSize) {
        return	$http.get('../specification/findPage/'+pageNum+'/'+pageSize);
    };
    //根据ID查询
    this.findById=function (id) {
        return $http.get("../specification/findById/"+id)
    };
    //保存与修改操作
    this.add=function (entity) {
        return $http.post("../specification/save",entity);
    };
    this.updata=function (entity) {
        return   $http.post("../specification/updata",entity);
    };
    //删除操作
    this.del=function (ids) {
        return	$http.get('../specification/del/'+ids);
    };
    this.specIdsList=function () {
        return $http.get('../specification/specification')
    }
});